### PR TITLE
add: list geospatial features

### PR DESCRIPTION
**Pull request by tuanng-cognite**

### CODE REVIEW

File: cognite/client/_api/geospatial.py [hunk 1]
```diff
@@ -491,6 +491,88 @@ def update_features(
...
+
+        This method allows to filter all features.
+
+        Args:
+            feature_type_external_id: the feature type to list features for
+            filter (Dict[str, Any]): the list filter
+            limit (int, optional): Maximum number of features to return. Defaults to 25. Set to -1, float("inf") or None
+                to return all features.
+            properties (Dict[str, Any]): the output property selection
+            allow_crs_transformation: If true, then input geometries will be transformed into the Coordinate Reference
```
Comment (vincent-cognite) [inline cognite/client/_api/geospatial.py:513]: Might be a copy paste error here. There are no input geometries with the /list endpoint

File: tests/tests_integration/test_api/test_geospatial.py [hunk 2]
```diff
@@ -449,6 +449,10 @@ def test_stream_features(self, cognite_client, large_feature_type, many_features
         feature_list = FeatureList(list(features))
         assert len(feature_list) == len(many_features)
 
+    def test_list(self, cognite_client, test_feature_type, test_features):
```
Comment (vincent-cognite) [inline tests/tests_integration/test_api/test_geospatial.py:452]: does this test the cursor mechanism?

Comment (tuanng-cognite) [reply tests/tests_integration/test_api/test_geospatial.py:452]: No, I'll improve the test

**Comment by tuanng-cognite** [comment]
@vincent-cognite see the changes [here](https://github.com/cognitedata/cognite-sdk-python/pull/1018/commits/7daebe9f0e8d9982721d282f286a203a92c2ae1a)

**Review by vincent-cognite — APPROVED** [review]

**Comment by tuanng-cognite** [comment]
@cognitedata/starbase @cognitedata/data-integration Please have a review

**Review by ponnapz — APPROVED** [review]

**Pull request merged**